### PR TITLE
[CDAP-14632] Remove Metadata upgrade code

### PR DESCRIPTION
**Pull request by anew**
best to review commit by commit.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataService.java [hunk 1]
```diff
@@ -83,12 +78,6 @@ protected void startUp() throws Exception {
       .build();
 
     httpService.start();
-
-    // Only first instance will run the migration thread.
-    if (Boolean.valueOf(cConf.get(Constants.Dataset.Executor.IS_UPGRADE_NEEDED, "false"))) {
```
Comment (CuriousVini) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataService.java:88]: Remove this constant as well?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java [hunk 2]
```diff
@@ -194,9 +191,6 @@
   private static final Gson GSON = new GsonBuilder()
     .registerTypeAdapter(NamespacedEntityId.class, new NamespacedEntityIdCodec())
     .create();
-  private static final Gson V1_GSON = new GsonBuilder()
-    .registerTypeAdapter(NamespacedEntityId.class, new NamespacedIdCodec())
```
Comment (CuriousVini) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java:198]: Do we need `NamespacedIdCodec` class as the only usage of it getting removed?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java [hunk 3]
```diff
@@ -1230,134 +1190,12 @@ static SearchTerm from(@Nullable NamespaceId namespaceId, String rawTerm) {
...
-   *
-   * @param limit  Batch size for scan/delete.
-   * @return metadata entries with list of {@link KeyValue} where key is timestamp for history row other wise
-   * it is null, value is {@link Object} and list of rowkeys to be deleted.
-   */
-  public MetadataEntries scanFromV1Table(int limit) {
-    MetadataEntries scannedEntries = new MetadataEntries(new LinkedList<>(), new LinkedList<>());
-
-    // Try to scan value rows first
-    byte[] rowPrefix = MdsKey.getValueRowPrefix();
```
Comment (CuriousVini) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java:1256]: should we also remove `MdsKey` and `MdsHistoryKey` key classes?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java [hunk 4]
```diff
@@ -1230,134 +1190,12 @@ static SearchTerm from(@Nullable NamespaceId namespaceId, String rawTerm) {
...
-   *
-   * @param row history row.
-   * @return {@link KeyValue} key is timestamp from history key, value is {@link MetadataEntry}.
-   */
-  @Nullable
-  private KeyValue<Long, Object> convertFromV1History(Row row) {
-    byte[] rowKey = row.getRow();
-    // History rows does not store entity type in the key. So to get the entity, we will read the value, deserialize it.
-    // However, in v2 tables, we are going to add type in the history row key.
-    MetadataV1 metadata = V1_GSON.fromJson(row.getString(HISTORY_COLUMN), MetadataV1.class);
```
Comment (CuriousVini) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java:1319]: should we also remove `MetadataV1` and `MetadataEntries`?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java [hunk 5]
```diff
@@ -597,150 +480,36 @@ private void publishAudit(MetadataRecordV2 previous, MetadataRecordV2 additions,
...
-        DatasetProperties.builder().add(MetadataDatasetDefinition.SCOPE_KEY, scope.name()).build());
-      markUpgradeComplete(scope);
-    }
-  }
-
-  private static DatasetId getMetadataDatasetInstance(MetadataScope scope) {
-    return MetadataScope.USER == scope ? BUSINESS_METADATA_INSTANCE_ID : SYSTEM_METADATA_INSTANCE_ID;
-  }
-
+  // TODO: there should be no public accessor for the dataset
```
Comment (CuriousVini) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java]: is there any jira to track this `TODO`?

**Review by CuriousVini — COMMENTED** [review]
couple of comments.

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataService.java (hunk 1 above)
Comment (anew) [reply cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataService.java:88]: good point. I am surprised this is a Dataset constant...

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java (hunk 2 above)
Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java:198]: you are right, looks like this is the only usage

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java (hunk 3 above)
Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java:1256]: yes

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java (hunk 4 above)
Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java:1319]: oh yeah

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java (hunk 5 above)
Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java]: yes, CDAP-14587. Added to the TODO

**Comment by anew** [comment]
Thanks for the review @CuriousVini . I addressed your comments, please take another look.

**Review by CuriousVini — APPROVED** [review]
LGTM 👍

**Comment by anew** [comment]
Build passed here. Will squash and merge.

**Pull request merged**